gdb: add or1k and or1knd target support

fjullien opened the pull request:
This patch is pretty much a copy of the current
GDB 7.2 port of the openrisc processor.

It now supports or1knd (no delay slot target).
The default target is or1k (with delay slot).
You can change the target arch with:

(gdb) set architecture or1knd
The target architecture is assumed to be or1knd

Signed-off-by: Franck Jullien franck.jullien@gmail.com

fjullien committed:
gdb: add or1k and or1knd target support

This patch is pretty much a copy of the current
GDB 7.2 port of the openrisc processor.

It now supports or1knd (no delay slot target).
The default target is or1k (with delay slot).
You can change the target arch with:

(gdb) set architecture or1knd
The target architecture is assumed to be or1knd

Signed-off-by: Franck Jullien <franck.jullien@gmail.com>

jeremybennett commented:
On Thu, 2013-01-10 at 01:14 -0800, Franck Jullien wrote:

> This patch is pretty much a copy of the current
> GDB 7.2 port of the openrisc processor.
> 
> It now supports or1knd (no delay slot target).
> The default target is or1k (with delay slot).
> You can change the target arch with:
> 
> (gdb) set architecture or1knd
> The target architecture is assumed to be or1knd

Hi Franck,

Great work, but you need to update the manual as well.

Have you run regression on it. GDB regression results for OR32 are
pretty dreadful, but it would be useful to check that nothing has got
worse!

Once you have the manual updated, and assuming there is nothing
horrendous new in regression, then please merge.

Best wishes,

Jeremy

> Signed-off-by: Franck Jullien franck.jullien@gmail.com
> 
> ---
> 
> You can merge this Pull Request by running
>   git pull https://github.com/Franck79/or1k-src prepare_patch
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/openrisc/or1k-src/pull/3
> 
> Commit Summary
>       \* gdb: add or1k and or1knd target support 
> File Changes
>       \* M gdb/ChangeLog (6) 
>       \* M gdb/configure.tgt (27) 
>       \* A gdb/or1k-tdep.c (2981) 
>       \* A gdb/or1k-tdep.h (432) 
> Patch Links:
>       \* https://github.com/openrisc/or1k-src/pull/3.patch 
>       \* https://github.com/openrisc/or1k-src/pull/3.diff 

## 

Tel:      +44 (1590) 610184
Cell:     +44 (7970) 676050
SkypeID: jeremybennett
Email:   jeremy.bennett@embecosm.com
Web:     www.embecosm.com

fjullien committed:
or1k-tdep: add  _initialize_or1k_tdep prototype

Add _initialize_or1k_tdep prototype to avoid getting
a warning during compilation.

Signed-off-by: Franck Jullien <franck.jullien@gmail.com>
